invision-app/app/components/movieDetails/movieDetails.js

```diff
@@ -2,14 +2,19 @@ import React from 'react';
 import './movieDetails.scss';
 import PropTypes from 'prop-types';
 import { connect } from 'react-redux';
-export function MovieDetails({dmovie,displayMainPage}) {
+export function MovieDetails({displayMainPage,history}) {
 
     function getYear(date) {
         let year = date.split('-')[0];
         return year;
     }
-    
-    const movie = dmovie;
+    if (!history.location.state) {
+        history.push({
+          pathname: '/noMovieFound'
+        });
+        return null;
+      }
+    const movie = history.location.state.movie;
```

Comment:
You can use **useSelector** hook from react-redux to get movie from Redux state by **id**
```suggestion
    const movie = useSelector(state => state.movies.movies.find(movie => movie.id == id));
```